applications/FluidDynamicsApplication/tests/cpp_tests/test_fluid_dynamics_constitutive_laws.cpp

```diff
@@ -0,0 +1,324 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+// System includes
+#include <set>
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "includes/model_part.h"
+#include "includes/cfd_variables.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+#include "custom_constitutive/euler_2d_law.h"
+#include "custom_constitutive/euler_3d_law.h"
+#include "custom_constitutive/newtonian_2d_law.h"
+#include "custom_constitutive/newtonian_3d_law.h"
+
+namespace Kratos {
+	namespace Testing {
+
+	    /** 
+	     * Checks the Newtonian fluid 2D constitutive law.
+	     */
+	    KRATOS_TEST_CASE_IN_SUITE(Newtonian2DConstitutiveLaw, FluidDynamicsApplicationFastSuite)
+		{
+            // Declare the constitutive law pointer as well as its required arrays
+            const unsigned int strain_size = 3;
+            Newtonian2DLaw::Pointer p_cons_law(new Newtonian2DLaw());
+            Vector stress_vector = ZeroVector(strain_size);
+            Vector strain_vector = ZeroVector(strain_size);
+            Matrix c_matrix = ZeroMatrix(strain_size, strain_size);
+
+            // Create a raw model part
+			ModelPart model_part("Main");
+			model_part.SetBufferSize(3);
+
+			// Variables addition
+			model_part.AddNodalSolutionStepVariable(DYNAMIC_VISCOSITY);
+
+			// Set the element properties
+			Properties::Pointer p_elem_prop = model_part.pGetProperties(0);
+			p_elem_prop->SetValue(DYNAMIC_VISCOSITY, 3.0e-01);
+			p_elem_prop->SetValue(CONSTITUTIVE_LAW, p_cons_law);
+
+			// Element creation
+			model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+			model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+			model_part.CreateNewNode(3, 0.0, 1.0, 0.0);
+			std::vector<ModelPart::IndexType> elemNodes {1, 2, 3};
+			model_part.CreateNewElement("Element2D3N", 1, elemNodes, p_elem_prop);
+			Element::Pointer p_element = model_part.pGetElement(1);
```

Comment:
This block is repeated in all tests (and poorly indented). Can this be moved to a function?

Reply:
I thought about that once I finished the test. It should be easy to do it.